pulsar-client-cpp/lib/PatternMultiTopicsConsumerImpl.cc

```diff
@@ -55,8 +55,9 @@ void PatternMultiTopicsConsumerImpl::autoDiscoveryTimerTask(const boost::system:
         return;
     }
 
-    if (state_ != Ready) {
-        LOG_ERROR("Error in autoDiscoveryTimerTask consumer state not ready: " << state_);
+    const auto state = state_.load();
+    if (state != Ready) {
+        LOG_ERROR("Error in autoDiscoveryTimerTask consumer state not ready: " << state);
```

Comment:
This change is redundant. `if (state_ != Ready)` is equivalent to `if (state_.load() != Ready)`, see https://en.cppreference.com/w/cpp/atomic/atomic/operator_T.

Usually when `load()` is called once, we can simplify

```c++
const auto x = atomicValue.load();
if (x == expectedValue) { /* ... */ }
```

to

```c++
if (atomicValue == expectedValue) { /* ... */ }
```

Reply:
Oh, there is a log output `state` in the 60th line here, I want to ensure their consistency

Reply:
I see. It makes sense.